client/src/components/SelectionField/services.js

```diff
@@ -0,0 +1,75 @@
+import { rethrowSimple } from "utils/simple-error";
+
+import { GalaxyApi } from "@/api";
+
+export async function getDataset(query, historyId) {
+    const { data, error } = await GalaxyApi().GET("/api/datasets", {
+        params: {
+            query: {
+                history_id: historyId,
+                q: ["name-contains"],
+                qv: [query],
+                offset: 0,
+                limit: 50,
+            },
+        },
+    });
+    if (error) {
+        rethrowSimple(error);
+    }
+    return data;
+}
+
+export async function getDatasetCollection(query, historyId) {
+    const { data, error } = await GalaxyApi().GET("/api/histories/{id}/contents", {
```

Comment:
The endpoint in the schema definition is "/api/histories/{history_id}/contents." Is there a specific reason why TypeScript wasn't used for this file? Converting this file to TypeScript would require some additional changes to the `SelectionField` file for improved type clarity, which I have suggested.